cmd/sync/main.go

```diff
@@ -132,8 +135,11 @@ func main() {
 				for _, ip := range ips {
 					backend := fmt.Sprintf("%v:%v", ip, upstream.Port)
 					upsServers = append(upsServers, nginx.StreamUpstreamServer{
-						Server:   backend,
-						MaxFails: 1,
+						Server:      backend,
+						MaxConns:    setPositiveIntOrDefault(upstream.MaxConns, 0),
+						MaxFails:    setPositiveIntOrDefault(upstream.MaxFails, 1),
```

Comment:
Sorry I missed this in the previous review. MaxFails can be 0 (this means is disabled) per official NGINX doc: `By default, the number of unsuccessful attempts is set to 1. The zero value disables the accounting of attempts.`  With the current implementation, the user can't set it to 0 to disable it.

So we need a different method for `max_fails` (I also suggest to remove `OrDefault` from every function to make them easier to read) and allow the user to sets a 0. However, because we don't know if the 0 is set by the user or the golang `ZeroValue` for int type, I suggest to change it to an `*int` instead of an `int`. This way we will have:

`setPositiveInt`, `setPositiveIntOrZeroFromPointer` and `setTime` (note how we got rid of `OrDefault`)

 What do you think? Maybe you have other ideas to cover this case.